src/main/java/world/bentobox/visit/panels/player/VisitPanel.java

```diff
@@ -824,6 +828,22 @@ private void updateFilter()
                 filter(island -> island.getMemberSet().stream().
                     map(User::getInstance).
                     anyMatch(User::isOnline)).
+                    filter(island -> {
+                        /* If essentials is not hooked, this filter doesn't apply */
+                        Essentials essentials = VisitAddon.getInstance().getEssentials();
+                        if (essentials == null) return true;
+
+                        /* If user can see vanished players, this filter does not apply */
+                        if (this.user.hasPermission("essentials.vanish.see")) return true;
+
+                        /* If the owner is online, and is vanished filter them out */
+                        OfflinePlayer owner = Bukkit.getOfflinePlayer(Objects.requireNonNull(island.getOwner()));
```

Comment:
Theoretically, this can be replaced with:
```
User owner = User.getInstance(island.getOwner);
```

Reply:
I'll refactor and re-test

Reply:
Refactored